Alignment/OfflineValidation/python/TkAlAllInOneTool/dataset.py

```diff
@@ -820,15 +826,60 @@ def dump_cff( self, outName = None, jsonPath = None, begin = None,
         theFile.close()
         return
 
-    def fileList( self, parent = False ):
+    def createdatasetfile_hippy(self, filename, filesperjob, firstrun, lastrun):
+        with open(filename, "w") as f:
+            for job in self.__chunks(self.fileList(firstRun=firstrun, lastRun=lastrun, forcerunselection=True), filesperjob):
+                f.write(",".join("'{}'".format(file) for file in job)+"\n")
+
+    @staticmethod
+    def getrunnumberfromfilename(filename):
+        parts = filename.split("/")
+        result = error = None
+        if parts[0] != "" or parts[1] != "store":
+            error = "does not start with /store"
+        elif parts[2] in ["mc", "relval"]:
+            result = 1
+        elif parts[-2] != "00000" or not parts[-1].endswith(".root"):
```

Comment:
Good question... I didn't know about it until @usarica told me.  It's true for many datasets but not for [this one](https://cmsweb.cern.ch/das/request?input=file%20dataset%3D/NoBPTX/Run2016D-TkAlCosmicsInCollisions-12Aug2016-v1/ALCARECO&instance=prod/global&idx=0&limit=10).  Actually Ulascan mentioned that in that case there are multiple runs in the same data file.

Basically at this point I was trying to be as strict as possible, and not remove the filename if it's not in the exact pattern that seems to be satisfied for most datasets.

If you have a better way of figuring out the run number from the filename that would be great.  This way is more efficient than using a lumi filter which requires opening and closing all the files.  It's particularly important for HipPy because we loop through the files multiple times, but since I was implementing it anyway I figured we might as well use it for validation too.